src/libcollections/string.rs

```diff
@@ -100,6 +101,31 @@ impl String {
         String { vec: ::slice::SliceExt::to_vec(string.as_bytes()) }
     }
 
+    #[cfg(not(stage0))]
+    /// Creates a new string buffer from the given string.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let s = String::from_str("hello");
+    /// assert_eq!(s.as_slice(), "hello");
+    /// ```
+    #[inline]
+    #[unstable(feature = "collections",
+               reason = "needs investigation to see if to_string() can match perf")]
+    #[cfg(not(test))]
+    pub fn from_str(string: &str) -> String {
+        String { vec: <[_]>::to_vec(string.as_bytes()) }
+    }
+
+    // HACK: `impl [T]` is not available in cfg(test), use `::slice::to_vec` instead of
```

Comment:
Sadly, no. To avoid conflicting lang items during unit testing, I had to mark the `impl [T]` with `cfg(not(test))`. Therefore during testing the methods that were in `impl [T]` but not in `core::slice::SliceExt` need to be resupplied as free functions. These free functions are only ever available and used during testing - so they shouldn't leak into the public API. (I left a comment explaining this in libcollections/slice.rs - but you can't see that file here beacuse of the huge diff)